### ISSUE TITLE
Issues with the enrichment tool

### ISSUE THREAD

**Issue by ValWood** [geneontology/panther-enrichment#10]
Love the new front page. I thought I should check the enrichment tool again.
I tried an enrichment with the complete set of 456  fission proteins annotated to "mitotic cell cycle"

This was the result:

![mitotic cell cycle](https://user-images.githubusercontent.com/7359272/52635990-4a350580-2ec3-11e9-82fc-a6f6743dd280.jpg)

This is due to the none inclusion of the regulates relationship. ...

This is a  *really big* problem, it makes much more sense to include this relationship for enrichment if "regulation" is correctly or incorrectly annotated.
The number of indirect causally upstream annotations to "regulates" is a pervasive problem, but not large enough to subvert the output of an enrichment (and would usually do so in a positive way because the genes are intimately connected to the process enriched for, by affecting the process albeit indirectly).

The larger problem is that, often  you "regulate a sub-process" you are part of the process. We don't instantiate this in GO, but it has never really worried me because the majority of tools include "regulates" relationship by default.

I could provide many 1000 of examples. Here is one I was looking at earlier

Translation elongation factor eIF-5A;
is annotated to
positive regulation of translational elongation (IGI)
positive regulation of translational termination (ISA)

this means that eIF-5A would not enrich to translation......the downside of performing enrichment like this is detrimental for *every* list that I have tested.
Try it!

**Comment by lpalbou** [comment]
Hi @ValWood , I am unable to reproduce the "sticky" problem. 

1. Are you referring to the main GO page and you can not change for instance the BP/MF/CC or species ? Or are you referring from the page on pantherdb.org ?
2. What browser are you using and do you have a set of steps to reproduce it ?
3. At this time, the GO site performs only overrepresentation test (single small gene set), we do not allow enrichment analysis (single large gene set with associated expression values). To perform the enrichment test, you need to go to http://pantherdb.org/ directly

**Comment by ValWood** [comment]
Hi ,
I'm very confused.

I'm using chrome
I did an enrichment from here: http://geneontology.org/
I'm positive that, the first time I did this  (yesterday) I needed to select the 'type' of analyses I wanted to perform (at panther) . I selected Overrepresentation Test.

Now whenever I arrive, it runs this test. What if I want to change it?

**Comment by ValWood** [comment]
I would not worry about the "sticky" problem. I can't repeat what I saw....

**Comment by ValWood** [comment]
I tried using firefox and didn't see it either. It might be worth looking what happens when somebody uses for the first time...

**Comment by lpalbou** [comment]
@ValWood I tried on 2 different machines but wasn't able to see it, maybe someone was working on something at that time but the **expected behavior** when clicking on "Launch" from the go site is to arrive on this page where the "type" of analysis is already set to overrepresentation test:

<img width="823" alt="screen shot 2019-02-16 at 4 44 06 pm" src="https://user-images.githubusercontent.com/24249870/52906849-197a0680-320a-11e9-81ad-f42775c86455.png">

Concerning the closures used for enrichment, I have passed on the message and we'll see what can be done. In the next iteration of Panther site, it sounds like a good idea to have an option for that.

**Comment by ValWood** [comment]
Yeah I have no idea what I was doing. It would be good to be able to toggle "regulates" transitivity/closure though.
val

**Comment by huaiyumi** [comment]
Regulates relation was once in PANTHER many years ago. It caused tremendous confusion to the users and the results were hard to interpret. That is why we pulled it. It is possible that the ontology is better now, and we can certainly test it again. We will put it on the todo list.

**Comment by lpalbou** [comment]
I personally like the idea of a toggle to enable an enrichment on terms directly related to the genes and on the terms the genes regulate. We have been discussing it from time to time too

**Comment by ValWood** [comment]
It is a major problem because many genes that are parts of processes are also involved in regulation. Our current annotation conventions require that we annotate these  to "regulates".
Therefore much gets lost from enrichments and slims.

see
https://github.com/pombase/curation/issues/2245
for examples

**Comment by lpalbou** [comment]
Just to confirm an action item: would a toggle to enrich either on is_a/part_of (current behavior) OR regulates only would be an acceptable solution ?

**Comment by ValWood** [comment]
Toggle would be a good start. Personally I think "regulates" should be the default, and can demo this is (I have never seen a dataset where this would not be the case), ....but its a consortium decision. Although, it would match what other tools do....

**Comment by cmungall** [comment]
> Just to confirm an action item: would a toggle to enrich either on
> is_a/part_of (current behavior) OR regulates only would be an acceptable
> solution ?
>

I believe the  request is to include genes that are involved in regulation
of the process, *as well as* genes involved in the process

**Comment by lpalbou** [comment]
@cmungall yes but it seems this needs more discussion and at least for the moment to have both:
> Regulates relation was once in PANTHER many years ago. It caused tremendous confusion to the users and the results were hard to interpret. That is why we pulled it. It is possible that the ontology is better now, and we can certainly test it again. We will put it on the todo list.

**Comment by ValWood** [comment]
Yes if toggle

is_a/part_of
OR
is_a/part_of + regulates

**Comment by lpalbou** [comment]
OK, so you don't think there is a value in highlighting only the functions that are regulated by genes ? We could still have:

* option 1: is_a + part_of
* option 2: is_a + part_of + regulates
* option 3: regulates

**Comment by ValWood** [comment]
personally I don't think option 3 would be useful....
